pkg/store/proxy.go

```diff
@@ -344,6 +355,45 @@ func startStreamSeriesSet(
 
 	wg.Add(1)
 	go func() {
+		const (
+			// approximate the length of each label being about 20 chars, e.g. "k8s_app_metric0"
+			approxLabelLen = int64(10)
+
+			// approximate the size if chunks by having 120 bytes in each?
+			approxChunkLen = int64(120)
+		)
+
+		// buffer query sizes and process in chunks of 20MB or so
+		// to avoid hammering cache lines with atomic increments.
+		const querySizeBufferSize = 20 * 1000 * 1000
```

Comment:
Any benchmarks to show that this is really optimal?

Reply:
This is really just a "finger in the air" figure. Arbitrary number. Reasoning being: we don't care if we go over the limit +/-20MB since we are talking about limits of 500MB+ in real scenarios.

I don't know what kind of benchmarks we can do to show this is "optimal". What is "optimal" in this case? I'm open to suggestions if someone has any better ideas.